tests/BaseTest.php

```diff
@@ -7,19 +7,133 @@
 abstract class BaseTest extends TestCase
 {
     /**
-     * Make sure expectException always exists, even on PHPUnit 4
      * @param string      $exception
      * @param string|null $message
      */
-    public function expectException($exception, $message = null)
+    public function expectExceptionGuzzle($exception, $message = null)
     {
         if (method_exists($this, 'setExpectedException')) {
             $this->setExpectedException($exception, $message);
         } else {
-            parent::expectException($exception);
+            $this->expectException($exception);
             if (null !== $message) {
                 $this->expectExceptionMessage($message);
             }
         }
     }
+
+    public function expectWarningGuzzle()
+    {
+        if (method_exists($this, 'expectWarning')) {
+            $this->expectWarning();
+        } elseif (class_exists('PHPUnit\Framework\Error\Warning')) {
+            $this->expectExceptionGuzzle('PHPUnit\Framework\Error\Warning');
+        } else {
+            $this->expectExceptionGuzzle('PHPUnit_Framework_Error_Warning');
+        }
+    }
+
+    /**
+     * @param string $type
+     * @param mixed  $input
+     */
+    public function assertInternalTypeGuzzle($type, $input)
```

Comment:
This is interesting. Why do we need this? I assume for legacy reasons. Can't we just drop the really old versions instead? Or isn't there like an official trait or polyfill? Feels like there should be.

Also, can we add a comment here why/until which PHPUnit version do we need this?

Reply:
We can cleanup in v2. As for v1, this works. ;)